src/pal/src/map/virtual.cpp

```diff
@@ -910,6 +930,66 @@ static LPVOID VIRTUALReserveMemory(
 
     InternalEnterCriticalSection(pthrCurrent, &virtual_critsec);
 
+    // If this is a request for special executable (JIT'ed) memory then, first of all,
+    // try to get memory from the executable memory allocator to satisfy the request.
+    if (((flAllocationType & MEM_RESERVE_EXECUTABLE) != 0) && (lpAddress == NULL))
+    {
+        pRetVal = g_executableMemoryAllocator.AllocateMemory(MemSize);
```

Comment:
I chose int32_t because this allocator can handle only up 2GB of memory. But if you think it will be cleaner to use SIZE_T instead, I have no problem with making this change.